main.py

```diff
@@ -1,70 +1,93 @@
+
+    return name, friend_name, friend_relationship, friend_gender
+
+
+def keep_swimming(friend_name, friend_gender):
+    while True:
+        friends_length = input("How long did you know " + friend_gender + "? | ")
+        if valid_time(friends_length):
+            break
+        else:
+            print('What\'s that? Try giving me a known time unit. Years, Months, Days, etc. ')
+    friends_length = strip_identifiers(friends_length)
+
+    animal: str = input(
+        "Cute. " + friends_length + ". got it. That's not very long for a robot, but I respect that. "
+                                    "If you could be any animal what would you be? | ")
+    animal = strip_identifiers(animal)
+
+    while True:
+        animal_friend: str = input("Do you think that " + friend_name.title() + " would be the same animal?  (y/n) | ")
+        if animal_friend in ['y', 'n']:
+            break
+        else:
+            print('please only respond with \'y\' or \'n\'')
+    if animal_friend == 'y':
         animal_friend = animal
         print("Cute, you guys have so much in common.")
     else:
         print("lol. You guys aren't really friends if you wouldn't even be in the same animal family.")
-        animal_friend: str = input("What would they be? | ") ## need to incorporate gender better here
+        animal_friend: str = input("What would they be? | ")  ## need to incorporate gender better here
+
+    return friends_length, animal_friend, animal
 
-    think(name,friend_name,friend_relationship,friend_gender,friends_length,animal_friend,animal)
 
-def think(name,friend_name,friend_relationship,friend_gender,friends_length,animal_friend,animal):
+def think(name, friend_name, friend_relationship, friend_gender, friends_length, animal_friend, animal):
     print("Ok... let me think...")
-    time.sleep(1)
-    print("Thinking...")
-    time.sleep(1)
-    print("Thinking...")
-    time.sleep(1)
-    print("Thinking...")
+    for _ in range(3):
+        time.sleep(1)
+        print('Thinking...')
     time.sleep(3)
-    print("So you, "+name+"... and your "+friend_relationship+", "+friend_name+", have been friends for "+friends_length+".")
+    print(
+        "So you, " + name + "... and your " + friend_relationship + ", "
+        + friend_name + ", have been friends for " + friends_length + ".")
 
     if animal_friend == animal:
-        print ("Just two "+animal+ "s living it up in nature. That's all there is to it.")
+        print("Just two " + animal + "s living it up in nature. That's all there is to it.")
     else:
-        print ("Just a "+animal+" and a "+animal_friend+". You and "+friend_gender+". Cool. That's all I've got.")
+        print(
+            "Just a " + animal + " and a " + animal_friend + ". You and " + friend_gender + ". Cool. That's all I've got.")
 
     print("How did I do?")
 
-start()
-##    print("Great," + myName + '!')
-##--else:
-##  print ("Sorry for asking...")
+
+def main():
```

Comment:
main function here, notice each sub-function returns what info it gains, and takes in what info it needs. 

This way, `think` can still get all the info it needs, but you don't need to pass `name` and `friend_relationship` into `keep_swimming`. `keep_swimming` doesn't need it.